pkg/controllers/termination/terminate.go

```diff
@@ -65,11 +65,19 @@ func (t *Terminator) drain(ctx context.Context, node *v1.Node) (bool, error) {
 	// https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdown
 	nonCritical := []*v1.Pod{}
 	critical := []*v1.Pod{}
-	for _, pod := range pods {
-		if pod.Spec.PriorityClassName == "system-cluster-critical" || pod.Spec.PriorityClassName == "system-node-critical" {
-			critical = append(critical, pod)
+	for _, p := range pods {
+		if val, ok := p.Annotations[provisioning.KarpenterDoNotEvictPodAnnotation]; ok && val == "true" {
+			zap.S().Debugf("Node %s has a do-not-evict pod %s. Watiing to drain until all do-not-evict pods are terminated.", node.Name, p.Name)
```

Comment:
Watiing.

Perhaps
`Waiting to drain node %s,  pod %s has 'do-not-evict' annotation"`